### PR TITLE
chore: cleanup on catalog check for Gsheets

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Instead of using the string let's use enum provided in types for checking

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 1]
```diff
@@ -744,11 +744,11 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
           engine_information,
         },
       });
-    }
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:747]: is this PR changing indentation/block scope?

Comment (hughhhh) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:747]: yea i moved it inside of the if so i have context for `engine`

**Review by lyndsiWilliams — APPROVED** [review]
lgtm!

**Pull request merged**